src/pages/GiftAid/GiftAid.js

```diff
@@ -108,34 +114,34 @@ function GiftAid(props) {
 
   /**
    * Updates validation state for form fields
-   * @param childState
-   * @param name
+   * @param thisFieldsState
+   * @param thisFieldsName
    */
-  const setFieldValidity = (childState, name) => {
-    const prevStateField = fieldValidation[name];
-    const fieldUndefined = prevStateField === undefined;
-    const valueUndefined = typeof prevStateField !== 'undefined' && prevStateField.value === undefined;
-    const newValue = typeof prevStateField !== 'undefined' && prevStateField.value !== childState.value;
-    const newState = (fieldUndefined === false && newValue) || (valueUndefined === true || newValue);
-
+  const setFieldValidity = (thisFieldsState, thisFieldsName) => {
+    const thisFieldsPreviousState = fieldValidation[thisFieldsName];
+    const isFieldUndefined = thisFieldsPreviousState === undefined;
+    const isFieldValueUndefined = typeof thisFieldsPreviousState !== 'undefined' && thisFieldsPreviousState.value === undefined;
+    const isNewFieldValue = typeof thisFieldsPreviousState !== 'undefined' && thisFieldsPreviousState.value !== thisFieldsState.value;
+    const isUpdatedState = (isFieldUndefined === false && isNewFieldValue) || (isFieldValueUndefined === true || isNewFieldValue);
+    
     // set field validation for marketing consent fields if present
-    const marketingConsentFieldsChanged = fieldUndefined === false &&
-      (childState.fieldValidation !== prevStateField.fieldValidation);
+    const marketingConsentFieldsChanged = isFieldUndefined === false &&
+      (thisFieldsState.fieldValidation !== thisFieldsPreviousState.fieldValidation);
+
+    if ((thisFieldsPreviousState && isUpdatedState) || marketingConsentFieldsChanged === true) {
+      
+      // Update postcode regex is country select value has changed
+      if (thisFieldsName === 'country' && thisFieldsState.value !== thisFieldsPreviousState.value ){
+
+        // Switch regex patterns if GB or not
+        setCurrentPostcodePattern(thisFieldsState.value === 'GB' ? GBPostCodePattern : OverseasPostCodePattern);
+
+        // Update counter we're using to retrigger postcode validation
+        setPostcodeRevalidate(postcodeRevalidate + 1 );
+      }
 
-    if ((prevStateField && newState) || marketingConsentFieldsChanged === true) {
```

Comment:
So it turns out this did literally nothing; I'm guessing either something has changed in the background in the meantime, or it just never worked and wasn't tidied up.

We're doing some kind of similar thing [here](https://github.com/comicrelief/giftaid-react/blob/master/src/pages/GiftAid/utils/Utils.js#L243), quite possibly fixing this non-working code.